hack/tools/extractcheck/README.md

```diff
@@ -0,0 +1,5 @@
+This directory contains a linter for checking SSA Extract calls.
```

Comment:
Checking them for what? Correct usage? Not being used at all?

From the code I learned that you want to forbid the usage of https://pkg.go.dev/k8s.io/apimachinery/pkg/util/managedfields#ExtractInto (one function) and functions like https://pkg.go.dev/k8s.io/client-go/applyconfigurations/core/v1#ExtractEndpointsStatus (many).

Is that correct?

I suspect that forbidigo can do that thanks to https://github.com/ashanbrown/forbidigo/pull/21. If my hunch is correct, then this entire tool isn't needed.

Reply:
I think forbidigo works:
```
ERROR: pkg/v1/cronjob.go:35:9: use of `managedfields.ExtractInto` forbidden because "should not be used because managedFields was removed" (forbidigo)
ERROR: 	err := managedfields.ExtractInto(cronJob, typed.ParseableType{}, fieldManager, b, subresource)
ERROR: 	       ^
ERROR: test/integration/client/client_test.go:1187:33: use of `appsv1ac.ExtractDeployment` forbidden because "should not be used because managedFields was removed" (forbidigo)
ERROR: 				extractedDeployment, err := appsv1ac.ExtractDeployment(actual, fieldMgr)
ERROR: 				                            ^
```

When excluding test files, only my artificial `pkg/v1/cronjob.go` test case gets reported. Usage in generated files is fine, golangci-lint by default suppresses issues in those.

=> https://github.com/kubernetes/kubernetes/pull/119728